flang/test/Transforms/debug-module-1.f90

```diff
@@ -0,0 +1,39 @@
+! RUN: %flang_fc1 -emit-fir -debug-info-kind=standalone -mmlir --mlir-print-debuginfo %s -o - | \
+! RUN: fir-opt --cg-rewrite --mlir-print-debuginfo | fir-opt --add-debug-info --mlir-print-debuginfo | FileCheck %s
```

Comment:
Probably best to test the pass only.

Reply:
The `AddDebugInfo` pass now operates on `XDeclareOp` so we need to run `CodeGenRewrite` before to convert the `DeclareOp`.

Reply:
I was trying to convey the following:
-> If it is the `AddDebugInfo` pass that you want to test, then give it the input after `cg-rewrite` and test its output.
-> If it is the `CodeGen` changes that you want to test, then give it the input before `Codegen` and test its output.

I am guessing the LLVM IR changes helps it easier to review and hence it might be OK to add it to the Integration directory. 

Generally tests that specifically checks the changes are easier to review and also maintain. This way of writing helps to ensure that when some unrelated changes happen these tests are not affected. For testing end to end flows we should ideally set up a separate buildbot or move these to the llvm-testsuite/Fortran/Debug.

Reply:
I agree with the suggestion on using output of `cg-rewrite` to test the `AddDebugInfo`. My initial motivation for basing test on fortran source code was that it was more readable.

As action items,

I will update `AddDebugInfo` tests to be based on the output of `cg-rewrite`
I will add `codegen` tests when it is involved.
I will keep adding llvm ir tests in Integration for now.